src/families/stellar/js-getTransactionStatus.ts

```diff
@@ -32,86 +42,170 @@ const getTransactionStatus = async (
+  if (estimatedFees.lt(t.networkInfo?.baseFee || 0)) {
+    errors.transaction = new StellarFeeSmallerThanBase();
+    // Entered fee is smaller than recommended
+  } else if (estimatedFees.lt(t.networkInfo?.fees || 0)) {
+    warnings.transaction = new StellarFeeSmallerThanRecommended();
   }
 
-  if (!errors.amount && amount.eq(0)) {
-    errors.amount = new AmountRequired();
+  // Operation specific checks
+  if (t.operationType === "changeTrust") {
+    // Check asset provided
+    if (!t.assetCode || !t.assetIssuer) {
+      // This is unlikely
+      errors.transaction = new StellarAssetRequired("");
+    }
+
+    // Has enough native balance to add new trustline
+    if (nativeAmountAvailable.minus(BASE_RESERVE).lt(0)) {
+      errors.amount = new StellarNotEnoughNativeBalanceToAddTrustline();
+    }
+  } else {
+    // Payment
+    // Check recipient address
+    if (!t.recipient) {
+      errors.recipient = new RecipientRequired("");
+    } else if (!isAddressValid(t.recipient)) {
+      errors.recipient = new StellarInvalidAddress("");
+    } else if (a.freshAddress === t.recipient) {
+      errors.recipient = new InvalidAddressBecauseDestinationIsAlsoSource();
+    }
+
+    const recipientAccount = await getRecipientAccount(t.recipient);
+
+    // Check recipient account
+    if (!recipientAccount.id && !errors.recipient && !warnings.recipient) {
+      if (recipientAccount.isMuxedAccount) {
+        errors.recipient = new StellarMuxedAccountNotExist();
+      } else {
+        if (isAssetPayment) {
+          errors.recipient = destinationNotExistMessage;
+        } else {
+          warnings.recipient = destinationNotExistMessage;
+        }
+      }
+    }
+
+    // Asset payment
+    if (isAssetPayment) {
+      const asset = findSubAccountById(a, t.subAccountId || "");
+
+      if (asset === null) {
+        // This is unlikely
+        throw new StellarAssetNotFound();
+      }
+
+      // Check recipient account accepts asset
+      if (
+        recipientAccount.id &&
+        !errors.recipient &&
+        !warnings.recipient &&
+        !recipientAccount.assetIds.includes(`${t.assetCode}:${t.assetIssuer}`)
+      ) {
+        errors.recipient = new StellarAssetNotAccepted("", {
+          assetCode: t.assetCode,
+        });
+      }
+
+      const assetBalance = asset?.balance || new BigNumber(0);
+
+      // @ts-expect-error check spendableBalance property
+      maxAmount = asset?.spendableBalance || assetBalance;
+      amount = useAllAmount ? maxAmount : t.amount;
+      totalSpent = amount;
+
+      if (!errors.amount && amount.gt(assetBalance)) {
+        errors.amount = new NotEnoughBalance();
+      }
+    } else {
+      // Native payment
+      maxAmount = nativeAmountAvailable;
+      amount = useAllAmount ? maxAmount : t.amount || 0;
+
+      if (amount.gt(maxAmount)) {
+        errors.amount = new NotEnoughBalance();
+      }
+
+      // TODO: ??? do we need to include fee in total?
```

Comment:
Yes `totalSpent` is intended to reflect the total amount that is taken from the account.
It just needs to be used consistently in subsequent tests, which I think is the case in the current implementation.